golang.go

```diff
@@ -134,26 +134,26 @@ func (g *golangApplier) checkSpecialConditions(target *os.File) (uint8, []string
 		return AutoGenerated, nil, nil
 	}
 
-	// Go 1.17 compiler flags (e.g., `//go:build !windows`)
-	if strings.HasPrefix(line, "//go:") {
+	var buildTags []string
+
+	// Go 1.17 compiler flags (e.g., `//go:build !windows`), or old compiler
+	// flags (e.g., `// +build !windows`).
+	// checks for Package comments as per https://blog.golang.org/godoc-documenting-go-code
+	if strings.HasPrefix(line, "//go:") || strings.HasPrefix(line, "// +build") {
+		buildTags = append(buildTags, line)
+
 		// read next line too: (`// +build !windows`)
-		if buf2, _, err := reader.ReadLine(); err == nil && strings.HasPrefix(string(buf2), "// +build ") {
-			return CompilerFlags, []string{line, string(buf2)}, nil
+		for {
+			buf2, _, err := reader.ReadLine()
+			if err != nil {
+				break
+			}
+			if strings.HasPrefix(string(buf2), "// +build ") {
+				buildTags = append(buildTags, string(buf2))
+			}
```

Comment:
I think you want an else break case here?

Reply:
ah, yes, doesn't hurt I guess 😄